core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -679,4 +780,29 @@ private ExprValue exprYear(ExprValue date) {
     return new ExprIntegerValue(date.dateValue().getYear());
   }
 
+  private Boolean isTimeZoneValid(ZoneId zone) {
```

Comment:
Isn't every ZoneId a valid time zone identifier? Why do we need this validation?

If we do, needs unit tests to show which subset of `ZoneId`s is valid and which are invalid.

Reply:
The valid/invalid test is just to check if the timezone is outside of the range
+14 and - 13:59. Also needs validation if the input was actually a valid timezone, if not it should return null too. Current commit doesn't have it, being pushed soon.

Reply:
I'd like to follow-up on this. As far as I can see, in Java, every ZoneId instance is a valid time zone.

Reply:
```mysql> select convert_tz("2000-01-01 10:00:00", "+14:00", "+00:00");
+-------------------------------------------------------+
| convert_tz("2000-01-01 10:00:00", "+14:00", "+00:00") |
+-------------------------------------------------------+
| 1999-12-31 20:00:00                                   |
+-------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select convert_tz("2000-01-01 10:00:00", "+14:01", "+00:00");
+-------------------------------------------------------+
| convert_tz("2000-01-01 10:00:00", "+14:01", "+00:00") |
+-------------------------------------------------------+
| NULL                                                  |
+-------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select convert_tz("2000-01-01 10:00:00", "-13:59", "+00:00");
+-------------------------------------------------------+
| convert_tz("2000-01-01 10:00:00", "-13:59", "+00:00") |
+-------------------------------------------------------+
| 2000-01-01 23:59:00                                   |
+-------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select convert_tz("2000-01-01 10:00:00", "-14:00", "+00:00");
+-------------------------------------------------------+
| convert_tz("2000-01-01 10:00:00", "-14:00", "+00:00") |
+-------------------------------------------------------+
| NULL                                                  |
+-------------------------------------------------------+
1 row in set (0.00 sec)```
The validator is to validate the timezones that match with MySQL standard. Java can handle all timezones being accepted and converted, but it won't match MySQL standard.

Reply:
If that's the case, we need to rename this as `isValidMySqlTimeZoneId` is similar

Reply:
See my comment above where the function is being used.

Reply:
Function name updated to `isValidMySqlTimeZoneId`.